control-plane/stor-port/src/types/v0/store/nexus.rs

```diff
@@ -109,7 +109,40 @@ pub struct NexusSpec {
     #[serde(default)]
     /// Hosts allowed to access the nexus.
     pub allowed_hosts: Vec<HostNqn>,
+    #[serde(default)]
+    /// Map defining Child and its Rebuild specific information.
+    pub rebuild_state: HashMap<ChildUri, RebuildInfo>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, Eq, PartialEq)]
+/// Rebuild specific information for a child.
+pub struct RebuildInfo {
+    /// Reconciler needs to wait for specified duration before starting Full Rebuild.
+    pub wait_time: Option<Duration>,
+    /// Current Rebuild stage of the child.
+    pub stage: RebuildStage,
```

Comment:
1. Yes, we can operate without having state but with that approach we might have to timewait, start partial rebuild / Full rebuild within same function. there is more logic coming in next phase as well. Thought of in memory state how we do in dsp but that doesnt seem logical here as we go out of scope after every reconcile run. 
2. Rebuildstage is like an intent that control plane reconciler has for a child hence we write that to nexus Spec and carry out each stage on separate reconciler fn.
3. There will be one more reconciler coming which will monitor PartialRebuild child and will delete state if its complete.
4. In this phase we delete state for a child after removing child from children list. This will change in phase 2 with some modifications in volume count reconciler. We will have to add one more reconciler for FullRebuild state and check rebuild stats incase it comes back if want to switch to partial rebuild.
5. Hashmap is empty when there is no rebuild. Any child rebuild will have max 4 spec transactions to etcd. transactions are spread across 3 tcd cluster and rebuild phase depending on when node comes back and wait time.

Reply:
>  there is more logic coming in next phase as well
Let's add the logic when we need to not ahead of time, baby steps, and be very conscious that when we add things to etcd we can't then remove them easily :)

Reply:
Sure Tiago,Not adding them now. Its for phase 2.  I will be testing removing those keys properly. FullRebuild key gets removed fine. That i checked in etcd. For PartialRebuild need to add logic to track the progress :)

Reply:
We have removed stages from this PR. Will revisit its usage during the next phase if there is a need